### ISSUE TITLE
Add possibity to force installing dependency gems

### ISSUE THREAD

**Issue by mpapis** [rubygems/rubygems#403]
Please add `--force-dependencies` flag to always install latest dependencies version in `GEM_HOME`.

Follow up of #385, there is a need to (re)install all the dependencies when installing gem.

A simple example would be:

```
gem install rspec-expectations -v 2.11.0
gem install rspec -v 2.11.0
```

Will make `rspec` work with `rspec-expectations` version `2.11.0`.

Now starting with a fresh environment (no gems) we do:

```
gem install rspec -v 2.11.0
```

Will make `rspec` work with `rspec-expectations` version `2.11.3` (at time of writing).

Returning to the initial example I would like to run:

```
gem install rspec-expectations -v 2.11.0
gem install rspec -v 2.11.0 --force-dependencies
```

Which should make `rspec` work with `rspec-expectations` version `2.11.3` (at time of writing).

**Comment by copiousfreetime** [comment]
Checking to see if there is still interest in this. I can see the use case, and the behavior still exists as described.

This appears to be the exact opposite of `--conservative` (do not update dependencies if they satisfy).

**Comment by drbrain** [comment]
I think `gem install --update-dependencies` would be a better name than `--force-dependencies`.

I think the behavior of the inverse, `gem install --no-update-dependencies`, is more guessable than `gem install --no-force-dependencies` (which, to me, could also mean "don't install any dependencies" or "don't force-install dependencies").

**Comment by copiousfreetime** [comment]
`--update-dependencies` sounds like a good name to me too.

**Comment by ddd** [comment]
Yes there is definitely still interested in this on my part.

Sent from Outlook Mobile<https://aka.ms/blhgte>



On Mon, Jan 18, 2016 at 6:16 PM -0800, "Jeremy Hinegardner" <notifications@github.com<mailto:notifications@github.com>> wrote:


--update-dependencies sounds like a good name to me too.

-

**Comment by deivid-rodriguez** [comment]
Old issue, but sounds like a good idea to me.

**Comment by deivid-rodriguez** [comment]
Hello :wave:!

I had a look at this and I found that we already have a `minimal_deps` option that implements this behaviour. However, the `minimal_deps == false` behaviour was broken, so I fixed it in #4030 so that you can use `--no-minimal-deps` to get this.

**Comment by deivid-rodriguez** [comment]
The flag name suggested here, `--update-dependencies` is way better than the one I ended up using `--no-minimal-deps`. I only used `--no-minimal-deps` for now because it was the negation of an already existent flag. But I'll change it to `--update-depedencies` since it is much better.
